### ISSUE TITLE
12 tests fail

### ISSUE THREAD

**Issue by yurivict** [SheffieldML/GPy#1075]
```
    def __getattr__(attr):
        # Warn for expired attributes, and return a dummy function
        # that always raises an exception.
        import warnings
        import math
        try:
            msg = __expired_functions__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
    
            def _expired(*args, **kwds):
                raise RuntimeError(msg)
    
            return _expired
    
        # Emit warnings for deprecated attributes
        try:
            val, msg = __deprecated_attrs__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
            return val
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
>           raise AttributeError(__former_attrs__[attr])
E           AttributeError: module 'numpy' has no attribute 'float'.
E           `np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
E           The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E               https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

/usr/local/lib/python3.11/site-packages/numpy/__init__.py:313: AttributeError
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
Intializing with GP...
Fit Sparse GP...
Fit BGPLVM...
Done.
________________________________________________________________________________ TestMisc.test_warped_gp_log ________________________________________________________________________________
[gw0] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestMisc object at 0x3498b7e77b10>

    def test_warped_gp_log(self):
        """
        A WarpedGP with the log warping function should be
        equal to a standard GP with log labels.
        Note that we predict the median here.
        """
        self.setup()
    
        k = GPy.kern.RBF(1)
        Y = np.abs(self.Y)
        logY = np.log(Y)
        m = GPy.models.GPRegression(self.X, logY, kernel=k)
        m.optimize()
        preds = m.predict(self.X)[0]
    
        warp_k = GPy.kern.RBF(1)
        warp_f = GPy.util.warping_functions.LogFunction(closed_inverse=False)
        warp_m = GPy.models.WarpedGP(self.X, Y, kernel=warp_k, warping_function=warp_f)
        warp_m.optimize()
        warp_preds = warp_m.predict(self.X, median=True)[0]
    
        warp_k_exact = GPy.kern.RBF(1)
        warp_f_exact = GPy.util.warping_functions.LogFunction()
        warp_m_exact = GPy.models.WarpedGP(
            self.X, Y, kernel=warp_k_exact, warping_function=warp_f_exact
        )
>       warp_m_exact.optimize(messages=True)

GPy/testing/test_model.py:663: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
GPy/core/gp.py:679: in optimize
    ret = super(GP, self).optimize(optimizer, start, messages, max_iters, ipython_notebook, clear_after_finish, **kwargs)
/usr/local/lib/python3.11/site-packages/paramz/model.py:110: in optimize
    with VerboseOptimization(self, opt, maxiters=max_iters, verbose=messages, ipython_notebook=ipython_notebook, clear_after_finish=clear_after_finish) as vo:
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:110: in __init__
    self.exps = exponents(self.fnow, self.current_gradient)
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:38: in exponents
    exps = [np.abs(np.float(fnow)),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

attr = 'float'

    def __getattr__(attr):
        # Warn for expired attributes, and return a dummy function
        # that always raises an exception.
        import warnings
        import math
        try:
            msg = __expired_functions__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
    
            def _expired(*args, **kwds):
                raise RuntimeError(msg)
    
            return _expired
    
        # Emit warnings for deprecated attributes
        try:
            val, msg = __deprecated_attrs__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
            return val
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
>           raise AttributeError(__former_attrs__[attr])
E           AttributeError: module 'numpy' has no attribute 'float'.
E           `np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
E           The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E               https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

/usr/local/lib/python3.11/site-packages/numpy/__init__.py:313: AttributeError
___________________________________________________________________ TestMisc.test_multioutput_regression_with_normalizer ____________________________________________________________________
[gw0] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestMisc object at 0x3498b7e93a90>

    def test_multioutput_regression_with_normalizer(self):
        """
        Test that normalizing works in multi-output case
        """
        self.setup()
    
        # Create test inputs
        X = self.X
        Y1 = np.sin(X) + np.random.randn(*X.shape) * 0.2
        Y2 = -np.sin(X) + np.random.randn(*X.shape) * 0.05
        Y = np.hstack((Y1, Y2))
    
        mu, std = Y.mean(0), Y.std(0)
        m = GPy.models.GPRegression(X, Y, normalizer=True)
>       m.optimize(messages=True)

GPy/testing/test_model.py:163: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
GPy/core/gp.py:679: in optimize
    ret = super(GP, self).optimize(optimizer, start, messages, max_iters, ipython_notebook, clear_after_finish, **kwargs)
/usr/local/lib/python3.11/site-packages/paramz/model.py:110: in optimize
    with VerboseOptimization(self, opt, maxiters=max_iters, verbose=messages, ipython_notebook=ipython_notebook, clear_after_finish=clear_after_finish) as vo:
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:110: in __init__
    self.exps = exponents(self.fnow, self.current_gradient)
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:38: in exponents
    exps = [np.abs(np.float(fnow)),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

attr = 'float'

    def __getattr__(attr):
        # Warn for expired attributes, and return a dummy function
        # that always raises an exception.
        import warnings
        import math
        try:
            msg = __expired_functions__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
    
            def _expired(*args, **kwds):
                raise RuntimeError(msg)
    
            return _expired
    
        # Emit warnings for deprecated attributes
        try:
            val, msg = __deprecated_attrs__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
            return val
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
>           raise AttributeError(__former_attrs__[attr])
E           AttributeError: module 'numpy' has no attribute 'float'.
E           `np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
E           The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E               https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

/usr/local/lib/python3.11/site-packages/numpy/__init__.py:313: AttributeError
_________________________________________________________________________ TestGradient.test_multiout_regression_md __________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestGradient object at 0x1f08956c9390>

    def test_multiout_regression_md(self):
        import GPy
    
        np.random.seed(0)
    
        self.setup()
    
        N = 20
        N_train = 5
        D = 8
        noise_var = 0.3
    
        k = GPy.kern.RBF(1, lengthscale=0.1)
        x_raw = np.random.rand(N * D, 1)
    
        # dimension assignment
        D_list = []
        for i in range(2):
            while True:
                D_sub_list = []
                ratios = []
                r_p = 0.0
                for j in range(3):
                    ratios.append(np.random.rand() * (1 - r_p) + r_p)
                    D_sub_list.append(int((ratios[-1] - r_p) * 4 * N_train))
                    r_p = ratios[-1]
                D_sub_list.append(4 * N_train - np.sum(D_sub_list))
                if (np.array(D_sub_list) != 0).all():
                    D_list.extend([a + N - N_train for a in D_sub_list])
                    break
    
        cov = k.K(x_raw)
    
        k_r = GPy.kern.RBF(2, lengthscale=0.4)
        x_r = np.random.rand(D, 2)
        cov_r = k_r.K(x_r)
    
        cov_all = np.repeat(np.repeat(cov_r, D_list, axis=0), D_list, axis=1) * cov
        L = GPy.util.linalg.jitchol(cov_all)
    
        y_latent = L.dot(np.random.randn(N * D))
    
        x = np.zeros((D * N_train,))
        y = np.zeros((D * N_train,))
        x_test = np.zeros((D * (N - N_train),))
        y_test = np.zeros((D * (N - N_train),))
        indexD = np.zeros((D * N_train), dtype=int)
        indexD_test = np.zeros((D * (N - N_train)), dtype=int)
    
        offset_all = 0
        offset_train = 0
        offset_test = 0
        for i in range(D):
            D_test = N - N_train
            D_train = D_list[i] - N + N_train
            y[offset_train : offset_train + D_train] = y_latent[
                offset_all : offset_all + D_train
            ]
            x[offset_train : offset_train + D_train] = x_raw[
                offset_all : offset_all + D_train, 0
            ]
            y_test[offset_test : offset_test + D_test] = y_latent[
                offset_all + D_train : offset_all + D_train + D_test
            ]
            x_test[offset_test : offset_test + D_test] = x_raw[
                offset_all + D_train : offset_all + D_train + D_test, 0
            ]
            indexD[offset_train : offset_train + D_train] = i
            indexD_test[offset_test : offset_test + D_test] = i
            offset_train += D_train
            offset_test += D_test
            offset_all += D_train + D_test
    
        y_noisefree = y.copy()
        y += np.random.randn(*y.shape) * np.sqrt(noise_var)
        x_flat = x.flatten()[:, None]
        y_flat = y.flatten()[:, None]
    
        Mr, Mc, Qr, Qc = 4, 3, 2, 1
    
        m = GPy.models.GPMultioutRegressionMD(
            x_flat,
            y_flat,
            indexD,
            Xr_dim=Qr,
            kernel_row=GPy.kern.RBF(Qr, ARD=False),
            num_inducing=(Mc, Mr),
        )
>       m.optimize_auto(max_iters=1)

GPy/testing/test_model.py:1465: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
GPy/models/gp_multiout_regression_md.py:222: in optimize_auto
    self.optimize(max_iters=int(0.1*max_iters),messages=verbose)
GPy/core/gp.py:679: in optimize
    ret = super(GP, self).optimize(optimizer, start, messages, max_iters, ipython_notebook, clear_after_finish, **kwargs)
/usr/local/lib/python3.11/site-packages/paramz/model.py:110: in optimize
    with VerboseOptimization(self, opt, maxiters=max_iters, verbose=messages, ipython_notebook=ipython_notebook, clear_after_finish=clear_after_finish) as vo:
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:110: in __init__
    self.exps = exponents(self.fnow, self.current_gradient)
/usr/local/lib/python3.11/site-packages/paramz/optimization/verbose_optimization.py:38: in exponents
    exps = [np.abs(np.float(fnow)),
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

attr = 'float'

    def __getattr__(attr):
        # Warn for expired attributes, and return a dummy function
        # that always raises an exception.
        import warnings
        import math
        try:
            msg = __expired_functions__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
    
            def _expired(*args, **kwds):
                raise RuntimeError(msg)
    
            return _expired
    
        # Emit warnings for deprecated attributes
        try:
            val, msg = __deprecated_attrs__[attr]
        except KeyError:
            pass
        else:
            warnings.warn(msg, DeprecationWarning, stacklevel=2)
            return val
    
        if attr in __future_scalars__:
            # And future warnings for those that will change, but also give
            # the AttributeError
            warnings.warn(
                f"In the future `np.{attr}` will be defined as the "
                "corresponding NumPy scalar.", FutureWarning, stacklevel=2)
    
        if attr in __former_attrs__:
>           raise AttributeError(__former_attrs__[attr])
E           AttributeError: module 'numpy' has no attribute 'float'.
E           `np.float` was a deprecated alias for the builtin `float`. To avoid this error in existing code, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
E           The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
E               https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations

/usr/local/lib/python3.11/site-packages/numpy/__init__.py:313: AttributeError
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
Intializing with GP...
Fit Sparse GP...
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
why are your arrays not F order?
Fit BGPLVM...
Done.
____________________________________________________________________ TestGradient.test_multioutput_sparse_regression_1D _____________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestGradient object at 0x1f08956c3490>

    def test_multioutput_sparse_regression_1D(self):
        self.setup()
        X1 = np.random.rand(500, 1) * 8
        X2 = np.random.rand(300, 1) * 5
        X = np.vstack((X1, X2))
        Y1 = np.sin(X1) + np.random.randn(*X1.shape) * 0.05
        Y2 = -np.sin(X2) + np.random.randn(*X2.shape) * 0.05
        Y = np.vstack((Y1, Y2))
    
        k1 = GPy.kern.RBF(1)
        m = GPy.models.SparseGPCoregionalizedRegression(
            X_list=[X1, X2], Y_list=[Y1, Y2], kernel=k1
        )
>       assert m.checkgrad()
E       assert False
E        +  where False = <bound method Gradcheckable.checkgrad of <GPy.models.sparse_gp_coregionalized_regression.SparseGPCoregionalizedRegression object at 0x1f08a085fb90>>()
E        +    where <bound method Gradcheckable.checkgrad of <GPy.models.sparse_gp_coregionalized_regression.SparseGPCoregionalizedRegression object at 0x1f08a085fb90>> = <GPy.models.sparse_gp_coregionalized_regression.SparseGPCoregionalizedRegression object at 0x1f08a085fb90>.checkgrad

GPy/testing/test_model.py:1214: AssertionError
______________________________________________________________________________ TestGridModel.test_alpha_match _______________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_grid.TestGridModel object at 0x1f08925ff5d0>

    def test_alpha_match(self):
>       kernel = GPy.kern.RBF(input_dim=self.dim, variance=1, ARD=True)
E       AttributeError: 'TestGridModel' object has no attribute 'dim'

GPy/testing/test_grid.py:32: AttributeError
_____________________________________________________________________________ TestGridModel.test_gradient_match _____________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_grid.TestGridModel object at 0x1f08925c6e10>

    def test_gradient_match(self):
>       kernel = GPy.kern.RBF(input_dim=self.dim, variance=1, ARD=True)
E       AttributeError: 'TestGridModel' object has no attribute 'dim'

GPy/testing/test_grid.py:41: AttributeError
____________________________________________________________________________ TestGridModel.test_prediction_match ____________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_grid.TestGridModel object at 0x1f08932940d0>

    def test_prediction_match(self):
>       kernel = GPy.kern.RBF(input_dim=self.dim, variance=1, ARD=True)
E       AttributeError: 'TestGridModel' object has no attribute 'dim'

GPy/testing/test_grid.py:58: AttributeError
___________________________________________________________________________ TestKernelGradientContinuous.test_OU ____________________________________________________________________________
[gw3] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_kernel.TestKernelGradientContinuous object at 0x1f08955b8490>

    def test_OU(self):
>       k = GPy.kern.OU(self.D - 1, ARD=True)
E       AttributeError: 'TestKernelGradientContinuous' object has no attribute 'D'

GPy/testing/test_kernel.py:547: AttributeError
____________________________________________________________________________ TestMisc.test_warped_gp_cubic_sine _____________________________________________________________________________
[gw1] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestMisc object at 0x8745ec74350>

    def test_warped_gp_cubic_sine(self):
        """
        A test replicating the cubic sine regression problem from
        Snelson's paper. This test doesn't have any assertions, it's
        just to ensure coverage of the tanh warping function code.
        """
        self.setup()
        max_iters = 100
    
        X = (2 * np.pi) * np.random.random(151) - np.pi
        Y = np.sin(X) + np.random.normal(0, 0.2, 151)
        Y = np.array([np.power(abs(y), float(1) / 3) * (1, -1)[y < 0] for y in Y])
        X = X[:, None]
        Y = Y[:, None]
    
        warp_m = GPy.models.WarpedGP(
            X, Y
        )  # , kernel=warp_k)#, warping_function=warp_f)
        warp_m[".*\.d"].constrain_fixed(1.0)
        warp_m.optimize_restarts(
            parallel=False, robust=False, num_restarts=5, max_iters=max_iters
        )
        warp_m.predict(X)
        warp_m.predict_quantiles(X)
        warp_m.log_predictive_density(X, Y)
        warp_m.predict_in_warped_space = False
>       warp_m.plot()
E       AttributeError: 'WarpedGP' object has no attribute 'plot'

GPy/testing/test_model.py:695: AttributeError
----------------------------------------------------------------------------------- Captured stdout call ------------------------------------------------------------------------------------
Optimization restart 1/5, f = -4.090421330513394
Optimization restart 2/5, f = 41.82249770487413
Optimization restart 3/5, f = -5.817697611141668
Optimization restart 4/5, f = -102.65483353662455
Optimization restart 5/5, f = -91.50886529992613
____________________________________________________________ TestGradientMultioutputGPModel.test_MultioutputGP_gradobs_prod_mix _____________________________________________________________
[gw1] freebsd14 -- Python 3.11.9 /usr/local/bin/python3.11

self = <GPy.testing.test_model.TestGradientMultioutputGPModel object at 0x8745ecd9b50>

    def test_MultioutputGP_gradobs_prod_mix(self):
        """
        Testing gradient observing MultioutputGP model with a mix of kernel types.
        """
        self.setup()
        for D in range(2, 4):
            kerns = []
            for d in range(D):
                if d % 2 == 0:
                    k = GPy.kern.RBF(input_dim=1)
                else:
                    k = GPy.kern.StdPeriodic(input_dim=1, period=self.period)
                    k.period.constrain_fixed()
                kerns.append(k)
            kern = reduce(lambda k0, k1: k0 * k1, kerns)
            kern.randomize()
>           self.check_model(kern)

GPy/testing/test_model.py:1776: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <GPy.testing.test_model.TestGradientMultioutputGPModel object at 0x8745ecd9b50>, kern = <GPy.kern.src.prod.Prod object at 0x874698a5790>

    def check_model(self, kern):
        """
        Checks predictions, hyperparameter gradients, and gradients of predicted
        posterior means and variances for MultioutputGP models that incorporate
        observed latent function gradient information.
        """
    
        D = kern.input_dim
    
        X_list = []
        Y_list = []
        for i in range(D + 1):
            # sample inputs for either latent function or partial derivatives
            X_i = np.random.uniform(*self.bounds, size=(self.train_points, D))
            # output of latent function or partial derivatives
            Y_i = (self.f(X_i) if (i == 0) else self.df(X_i)[:, i - 1])[:, None]
            # noisy observations
            Y_i += np.random.normal(scale=self.noise_std, size=Y_i.shape)
    
            X_list.append(X_i)
            Y_list.append(Y_i)
    
        # the kernel is accompanied with derivative kernels, one for each dimension
        kernel_list = [kern] + [GPy.kern.DiffKern(kern, d) for d in range(D)]
    
        # create model and check its hyperparameter gradient
        likelihood_list = [GPy.likelihoods.Gaussian(variance=self.noise_std**2)] * (
            D + 1
        )
        model = GPy.models.MultioutputGP(X_list, Y_list, kernel_list, likelihood_list)
        model.likelihood.constrain_fixed()
        assert model.checkgrad(step=1e-3)
    
        # optimize the model, and check its hyperparameter gradient again
        model.optimize()
>       assert model.checkgrad(step=1e-3)
E       assert False
E        +  where False = <bound method Gradcheckable.checkgrad of <GPy.models.multioutput_gp.MultioutputGP object at 0x8746980e790>>(step=0.001)
E        +    where <bound method Gradcheckable.checkgrad of <GPy.models.multioutput_gp.MultioutputGP object at 0x8746980e790>> = <GPy.models.multioutput_gp.MultioutputGP object at 0x8746980e790>.checkgrad

GPy/testing/test_model.py:1671: AssertionError
```

Version: 1.13.1
py311-numpy-1.25.0_7
Python: 3.11
FreeBSD 14.0

**Comment by MartinBubel** [comment]
Hi @yurivict, 

could you please tell us what you are doing?
In the above, I see references to code that has been dropped/fixed as off v1.13.0. So I doubt you are running `pytest` on version v1.13.1.
Without furhter information from your side, there is few I can do.

**Comment by MartinBubel** [comment]
Gonna close this now as there is no activity/ answer. Feel free to reopen.
